test/extended/openstack/kuryr.go

```diff
@@ -0,0 +1,164 @@
+	"math/rand"
+	"strconv"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/subnets"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	testutils "k8s.io/kubernetes/test/utils"
+)
+
+var _ = g.Describe("[sig-installer][Feature:openstack][Kuryr] The OpenStack platform", func() {
+	var namespaces []*v1.Namespace
+	var networkClient *gophercloud.ServiceClient
+	var dc dynamic.Interface
+	var clientSet *kubernetes.Clientset
+
+	g.BeforeEach(func() {
+		g.By("preparing openshift dynamic client")
+		cfg, err := e2e.LoadConfig()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		dc, err = dynamic.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		clientSet, err = e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		c, err := configv1client.NewForConfig(cfg)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		network, err := c.ConfigV1().Networks().Get(context.Background(), "cluster", metav1.GetOptions{})
+
+		if network.Status.NetworkType != "Kuryr" {
+			e2eskipper.Skipf("Test can only be run with Kuryr SDN")
+		}
+
+		g.By("preparing openstack client")
+		networkClient, err = client(serviceNetwork)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+	})
+
+	g.AfterEach(func() {
+		clientSet, err := e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		g.By("Cleanup up Namespaces")
+		for _, ns := range namespaces {
+
+			o.Expect(clientSet.CoreV1().Namespaces().Delete(context.TODO(), ns.Name, metav1.DeleteOptions{})).To(o.Succeed())
+		}
+	})
+	g.It("should create a subnet when a pod without hostNetwork is created", func() {
+
+		g.By("Creating a Namespace and pods")
+		{
+
+			clientSet, err := e2e.LoadClientset()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			ns := CreateNamespace(clientSet, "kuryr-hostnetwork")
+
+			namespaces = append(namespaces, ns)
+			listOpts := subnets.ListOpts{
+				Name: "ns/" + ns.Name + "-subnet",
+			}
+			g.By("Creating a Pod with hostNetwork=true")
+			CreatePod(clientSet, ns.Name, "kuryr-hostnetwork", true)
+			o.Expect(len(ListSubnets(networkClient, listOpts))).To(o.Equal(0), "A subnet should not be created when a pod with Hostnetwork is created")
+			g.By("Creating a Pod with hostNetwork=false")
+			CreatePod(clientSet, ns.Name, "kuryr-nohostnetwork", false)
+			o.Expect(len(ListSubnets(networkClient, listOpts))).To(o.Equal(1), "A subnet should be created when a pod is created")
+		}
+	})
+})
+
+func RandomSuffix() string {
+	return strconv.Itoa(rand.Intn(10000))
+}
+
+func ListSubnets(networkClient *gophercloud.ServiceClient, listOpts subnets.ListOpts) []subnets.Subnet {
```

Comment:
Should all the functions below go to some utils file? Also I'm fairly sure there are methods for creation of pods in OpenShift tests, I wonder if we can reuse them. @pierreprinetti, maybe you know better on code organization and reusal?

Reply:
I moved to utils.go , I'm open to other proposals